tests/prague/eip7623_increase_calldata_cost/test_refunds.py

```diff
@@ -0,0 +1,289 @@
+        "tx_floor_data_cost is too low, there might have been a gas cost change that caused this "
+        "test to fail. Try increasing the intrinsic_gas_data_floor_minimum_delta fixture."
+    )
+
+    # Dumb for-loop to find the execution gas cost that will result in the expected refund.
+    while execution_gas_cost(execution_gas) < tx_floor_data_cost:
+        execution_gas += 1
+    if refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_EQUAL_TO_DATA_FLOOR:
+        return execution_gas
+    elif refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_GREATER_THAN_DATA_FLOOR:
+        return execution_gas + 1
+    elif refund_test_type == RefundTestType.EXECUTION_GAS_MINUS_REFUND_LESS_THAN_DATA_FLOOR:
+        return execution_gas - 1
+
+    raise ValueError("Invalid refund test type")
+
+
+@pytest.fixture
+def refund(
+    tx_intrinsic_gas_cost_before_execution: int,
+    execution_gas_used: int,
+    max_refund: int,
+) -> int:
+    """Return the refund gas of the transaction."""
+    total_gas_used = tx_intrinsic_gas_cost_before_execution + execution_gas_used
+    return min(max_refund, total_gas_used // 5)
+
+
+@pytest.fixture
+def to(
+    pre: Alloc,
+    execution_gas_used: int,
+    prefix_code: Bytecode,
+    prefix_code_gas: int,
+    code_storage: Dict,
+) -> Address | None:
+    """
+    Return a contract that consumes the expected execution gas.
+
+    At the moment we naively use JUMPDEST to consume the gas, which can yield very big contracts.
+
+    Ideally, we can use memory expansion to consume gas.
+    """
+    extra_gas = execution_gas_used - prefix_code_gas
+    return pre.deploy_contract(
+        prefix_code + (Op.JUMPDEST * extra_gas) + Op.STOP,
+        storage=code_storage,
+    )
+
+
+@pytest.fixture
+def tx_gas_limit(
+    tx_intrinsic_gas_cost_including_floor_data_cost: int,
+    tx_intrinsic_gas_cost_before_execution: int,
+    execution_gas_used: int,
+) -> int:
+    """
+    Gas limit for the transaction.
+
+    The gas delta is added to the intrinsic gas cost to generate different test scenarios.
+    """
+    tx_gas_limit = tx_intrinsic_gas_cost_before_execution + execution_gas_used
+    assert tx_gas_limit >= tx_intrinsic_gas_cost_including_floor_data_cost
+    return tx_gas_limit
+
+
+@pytest.mark.parametrize(
+    "refund_test_type",
+    [
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_GREATER_THAN_DATA_FLOOR,
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_LESS_THAN_DATA_FLOOR,
+        RefundTestType.EXECUTION_GAS_MINUS_REFUND_EQUAL_TO_DATA_FLOOR,
+    ],
+)
+@pytest.mark.parametrize(
+    "refund_type",
+    [
+        RefundType.STORAGE_CLEAR,
+        RefundType.STORAGE_CLEAR | RefundType.AUTHORIZATION_EXISTING_AUTHORITY,
+        RefundType.AUTHORIZATION_EXISTING_AUTHORITY,
+    ],
+)
+def test_gas_refunds_from_data_floor(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    tx: Transaction,
+    tx_floor_data_cost: int,
```

Comment:
looks like a lot of logic is hidden inside fixtures. the fixtures are quite implicit on input args makes me a bit confused compared to when we call methods directly from the test

Reply:
See response on https://github.com/ethereum/execution-spec-tests/pull/1071/files/42d6deea2b7eeba9378f1236617361680c25e399#r1927121971

I think an extension like this does make it easier to follow.